app/src/main/java/io/novafoundation/nova/app/root/presentation/RootViewModel.kt

```diff
@@ -55,6 +56,10 @@ class RootViewModel(
         syncCurrencies()
 
         updatePhishingAddresses()
+
+        walletConnectService.setOnPairErrorCallback {
+            showError(it)
+        }
```

Comment:
I would rather suggest to expose LiveData<Event<Error>> in `WalletConnectService` and to subscribe to it in `rootViewModel`